src/qt/nametablemodel.cpp

```diff
@@ -410,16 +413,42 @@ QVariant NameTableModel::data(const QModelIndex &index, int role) const
 
     NameTableEntry *rec = static_cast<NameTableEntry*>(index.internalPointer());
 
-    // TODO: implement Qt::ForegroudRole for font color styling for states?
+    // TODO: implement Qt::ForegroundRole for font color styling for states?
     // TODO: implement Qt::ToolTipRole show name status on tooltip
     if(role == Qt::DisplayRole || role == Qt::EditRole)
     {
         switch(index.column())
         {
             case Name:
-                return rec->name;
+            {
+                NameNamespace ns = NamespaceFromName(rec->name.toStdString());
+                std::string strDesc = DescFromName(rec->name.toStdString(), ns);
+                QString desc = QString::fromStdString(strDesc);
+
+                switch(ns)
+                {
+                    case NameNamespace::Domain:
+                        return tr("Domain %1").arg(desc);
+                    case NameNamespace::DomainData:
+                        return tr("Domain data %1").arg(desc);
+                    case NameNamespace::Identity:
+                        return tr("Identity %1").arg(desc);
+                    case NameNamespace::IdentityData:
+                        return tr("Identity data %1").arg(desc);
+                    case NameNamespace::NonStandard:
+                        return tr("Non-standard name %1").arg(desc);
```

Comment:
Out of curiosity, I remember already seeing code like this, but I can't find it in the current codebase.  Was that just another PR or am I simply mistaken here?  Of course if we already have a switch to generate text representations, we should reuse it, but seems we don't, so this is fine.

Reply:
There's something mildly similar in `transactiontablemodel.cpp`, but it's for generating a text representation of the namespace *without* the rest of the name. That might be what you were thinking of?